credential.go

```diff
@@ -21,6 +22,13 @@ type Credential struct {
 	nonrevCache chan *NonRevocationProofBuilder
 }
 
+// Statement that relevant attribute satisfies a*m-k > 0, and that a*m-k can be split into squares with given splitter
+type RangeStatement struct {
+	a     int
+	k     *big.Int
```

Comment:
The fact that these are unexported means that it is impossible for `irmago` or other calling packages to use the entire range proofs functionality provided by this PR, as they cannot construct instances of this to pass to `CreateDisclosureProofBuilder`.

At least, then, `a` and `k` will need to be exported. In addition, for callers it will help if you rename them to something more descriptive, as these two letters are essentially arbitrary. Perhaps:
* `a` -> `Factor`
* `k` -> `Bound`?

As to the `split` parameter, would it be possible to make this optional and to use four-splits when it is not specified? Probably the VWS usecase will be the only one that needs three-splits.

Reply:
fixed (including default on split)